java-frontend/src/main/java/org/sonar/java/se/checks/OptionalGetBeforeIsPresentCheck.java

```diff
@@ -47,7 +47,12 @@
     "\"NoSuchElementException\" will be thrown when invoking method \"%s()\" without verifying Optional parameter.");
 
   private enum OptionalConstraint implements Constraint {
-    PRESENT, NOT_PRESENT
+    PRESENT, NOT_PRESENT;
+
+    @Override
+    public boolean hasPreciseValue() {
+      return this == NOT_PRESENT;
```

Comment:
Please discard this comment. I do agree that `PRESENT` is not precise. I was thinking only to cases where we consider always the same symbolic value. When we are in presence of different symbolic values, `NOT_PRESENT` guarantees invariance, while `PRESENT` does not.